var/spack/repos/builtin/packages/npb/package.py

```diff
@@ -22,9 +22,19 @@
 # License along with this program; if not, write to the Free Software
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
 ##############################################################################
+import numbers
+
 from spack import *
 
 
+def is_integral(x):
```

Comment:
I like the idea of values being (able to be) a lamba (that seems to be the pythonic word...) that tests a value and returns True/False.  Given that, having&supporting a second knob named `type` seems redundant->more_work->confusing. 

You could think about what Perl/Moose did with [TypeConstraints](http://search.cpan.org/~ether/Moose-2.2005/lib/Moose/Util/TypeConstraints.pm), it's a fairly flexible system for adding "types", I'd imagine you might want something like that so that I could say "type=IntBetween23And42" or "type=PhoneNumber" or ....

So far we have one example of this and I fear over-engineering it.

RE a naming standard:

- What does `any_integer` do?  It returns True/False if its arg is an integer.
- What does `is_integral/is_integer` do?  It returns True/False if its arg is an integer.

I think that the second is more mnemonic/descriptive, but that's probably because the "is_" bit implies a boolean return value and perhaps that's not critical.  In fact, you're going to end up with a library of functions, all of which begin with `is_` (or `is_not_`), at which point it might loose it's value.

Any votes for `integerp`? (mostly :wink:)

Bikeshed?